File: airpulse-ui/src/app/pages/admin/services/employee.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { BaseService } from "../../shared/services/base.service";
import { LoaderService } from "../../shared/services/loader.service";
import { EmployeeRequest, EmployeeResponse } from "../model/employee.model";

@Injectable({
    providedIn: 'root',
})
export class EmployeeService extends BaseService<EmployeeResponse, EmployeeRequest> {
    constructor(protected override http: HttpClient, protected override loader: LoaderService) {
        super(http, 'emp/employee', loader);
    }
}

File: employee-management/employee-model/src/main/java/com/pulse/air/employee/contract/EmployeeService.java
package com.pulse.air.employee.contract;

import com.pulse.air.common.model.BaseSearchRequest;
import com.pulse.air.commons.contract.BaseCRUDService;
import com.pulse.air.employee.model.employee.EmployeeRequest;
import com.pulse.air.employee.model.employee.EmployeeResponse;

public interface EmployeeService extends BaseCRUDService<EmployeeResponse, EmployeeRequest, BaseSearchRequest> {

}
